sdk/test-utils/recorder-new/MIGRATION.md

```diff
@@ -0,0 +1,278 @@
+  },
+});
+```
+
+Other sanitizers for more complex use cases are also available.
+
+## AAD and the new `NoOpCredential`
+
+The new recorder does not record AAD traffic at present. As such, tests with clients using AAD should make use of the new `@azure-tools/test-credential` package, installed as follows:
+
+```bash
+$ rush add --dev --caret -p @azure-tools/test-credential
+```
+
+This package provides a `NoOpCredential` implementation of `TokenCredential` which makes no network requests, and should be used in playback mode. The provided `createTestCredential` helper will handle switching between NoOpCredential in playback and ClientSecretCredential when recording for you:
+
+```ts
+const credential = createTestCredential();
+
+// Create your client using the test credential.
+new MyServiceClient(<endpoint>, credential);
+```
+
+Since AAD traffic is not recorded by the new recorder, there is no longer a need to remove AAD credentials from the recording using a sanitizer.
+
+## Browser tests and modifications to Karma configuration
+
+When running browser tests, the recorder relies on an environment variable to determine where to save the recordings. Add this snippet to your `karma.conf.js`:
+
+```ts
+const { relativeRecordingsPath } = require("@azure-tools/test-recorder-new");
+
+process.env.RECORDINGS_RELATIVE_PATH = relativeRecordingsPath();
+```
+
+And then, again in `karma.conf.js`, add the variable to the list of environment variables:
+
+```ts
+module.exports = function (config) {
+  config.set({
+    /* ... */
+
+    envPreprocessor: [
+      ,
+      /* ... */ "RECORDINGS_RELATIVE_PATH", // Add this!
+    ],
+
+    /* ... */
+  });
+};
+```
+
+The following configuration options in `karma.config.js` should be **removed**:
+
+```ts
+browserConsoleLogOptions: {
+  terminal: !isRecordMode(),
+}
+
+/* ... */
+
+jsonToFileReporter: {
+  filter: jsonRecordingFilterFunction,  outputPath: ".",
+}
+```
+
+## Changes to `ci.yml`
+
+You must set the `TestProxy` parameter to `true` to enable the test proxy server in your SDK's `ci.yml` file.
+
+```yaml
+# irrelevant sections of ci.yml omitted
+
+extends:
+  template: ../../eng/pipelines/templates/stages/archetype-sdk-client.yml
+  parameters:
+    TestProxy: true # Add me!
+```
+
+## Migrating your recordings
+
+Once you have made the necessary code changes, it is time to re-record your tests using the Unified Recorder to complete the migration. To do this, first **delete** the directory containing the old recordings (the `recordings` folder). Then, run your tests with the `TEST_MODE` environment variable to `record`.
+
+If everything succeeds, the new recordings will be made available in the `recordings` directory. Inspect them to make sure everything looks OK (no secrets present, etc.), and then run the tests in playback mode to ensure everything is passing. If you're running into issues, check out the [Troubleshooting section](#troubleshooting).
```

Comment:
troubleshooting is empty?

Reply:
uh oh...

why do I remember writing it if it's not there...